openedx/core/lib/teams_config.py

```diff
@@ -332,3 +360,63 @@ def _clean_max_team_size(value):
     if value < 0:
         return None
     return value
+
+
+def create_team_set_partition_with_course_id(course_id, team_sets=None):
+    """
+    Create and return the dynamic enrollment track user partition based only on course_id.
+    If it cannot be created, None is returned.
+    """
+    if not team_sets:
+        team_sets = _get_team_sets(course_id) or {}
+
+    try:
+        team_scheme = UserPartition.get_scheme(TEAM_SCHEME)
+    except UserPartitionError:
+        log.warning(f"No {TEAM_SCHEME} scheme registered, TeamUserPartition will not be created.")
+        return None
+
+    # Get team-sets from course and create user partitions for each team-set
+    # Then get teams from each team-set and create user groups for each team
+    partitions = []
+    for team_set in team_sets:
+        partition = team_scheme.create_user_partition(
+            id=team_set.dynamic_user_partition_id,
+            name=_(f"Team Group: {team_set.name}"),  # lint-amnesty, pylint: disable=translation-of-non-string
```

Comment:
pylint is yelling at you here because you can't format a string within `_(...)`. That's because it would try to translate the formatted string, which of course could be any number of indeterminate strings:
* `"Team Group: Homework 1"`
* `"Team Group: Project 2"`
* `"Team Group: Another teamset"`
* etc... 

Instead, we want to send the *template* to Transifex (`"Team Group: {team_set_name}"`). Translators will see the template, they will translate it, and then our edx-platform app can format using the translated template at runtime.

We can achieve this by using classic string formatting instead of an f-string:
```suggestion
            name=_("Team Group: {team_set_name}").format(team_set_name=team_set.name),
```